presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsCalculator.java

```diff
@@ -113,6 +119,44 @@ protected SymbolStatsEstimate visitLiteral(Literal node, Void context)
             return estimate.build();
         }
 
+        @Override
+        protected SymbolStatsEstimate visitFunctionCall(FunctionCall node, Void context)
+        {
+            Map<NodeRef<Expression>, Type> expressionTypes = getExpressionTypes(session, node, types);
+            ExpressionInterpreter interpreter = ExpressionInterpreter.expressionOptimizer(node, metadata, session, expressionTypes);
+            Object value = interpreter.optimize(NoOpSymbolResolver.INSTANCE);
+
+            if (value == null || value instanceof NullLiteral) {
+                return nullStatsEstimate();
+            }
+
+            if (value instanceof Expression) {
+                // value is not a literal
```

Comment:
Really?

```
public abstract class Literal
        extends Expression
```

I think you meant that function call was not optimized to constant.